### PR TITLE
Add CLI option to ignore invokedynamic types

### PR DESCRIPTION
**Pull request by lukebemish**
I've recently ran up against #26 in a project of mine - I get so much log spam due to tiny-remapper finding groovy indy instructions that it cannot remap that it is impossible to see any actual issues in the logs. I would like some way of turning off that warning for specific indy instructions, similar to what is already done for string concatenation. This PR adds a system property, `net.fabricmc.tinyremapper.knownindybsm` (this could definitely change, I'm by no means sold on it) for providing a list of instructions that tiny-remapper can safely assume do not need to be remapped.

If you think there may be a better approach to this than a system property, please let me know, and I can see about implementing that.

### CODE REVIEW

File: src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java [hunk 1]
```diff
@@ -277,12 +279,15 @@ public void visitInvokeDynamicInsn(String name, String descriptor, Handle bootst
 					bootstrapMethodArguments);
 		}
 
+		private static final Set<String> ALLOWED_INDY_BSM = Stream.of(System.getProperty("net.fabricmc.tinyremapper.knownindybsm", "").split(",")).collect(Collectors.toSet());
```
Comment (modmuss50) [inline src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java:282]: This should be set via the CLI and via the TinyRemapper.Builder. See how `forcePropagation` is set.

File: src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java [hunk 2]
```diff
@@ -277,12 +279,15 @@ public void visitInvokeDynamicInsn(String name, String descriptor, Handle bootst
...
+
 		private static Handle getLambdaImplementedMethod(String name, String desc, Handle bsm, Object... bsmArgs) {
 			if (isJavaLambdaMetafactory(bsm)) {
 				assert desc.endsWith(";");
 				return new Handle(Opcodes.H_INVOKEINTERFACE, desc.substring(desc.lastIndexOf(')') + 2, desc.length() - 1), name, ((Type) bsmArgs[0]).getDescriptor(), true);
 			} else if (bsm.getOwner().equals("java/lang/invoke/StringConcatFactory")
-					|| bsm.getOwner().equals("java/lang/runtime/ObjectMethods")) {
```
Comment (modmuss50) [inline src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java:285]: I think it would also be best to move these into the set.

**Review by modmuss50 — CHANGES_REQUESTED** [review]

**Comment by modmuss50** [comment]
Another option with this is to just not log issues at all, or maybe log it once (when a flag is set?)

**Comment by lukebemish** [comment]
Alright, I *think* I hooked that up to the CLI system in a way that makes sense; let me know if not. I'm not quite sure enough about how TR works to try poking at adding the "warn only once" system, but that could always be added later as well.

**Review by modmuss50 — APPROVED** [review]

File: src/main/java/net/fabricmc/tinyremapper/Main.java [hunk 3]
```diff
@@ -20,8 +20,11 @@
 
 import java.io.BufferedReader;
 import java.io.File;
+import java.io.FileInputStream;
 import java.io.FileReader;
```
Comment (modmuss50) [inline src/main/java/net/fabricmc/tinyremapper/Main.java:24]:
```suggestion
```
Removing this unused import.

**Pull request merged**